pkg/util/kubernetes/apiserver/openshift.go

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubeapiserver
+
+package apiserver
+
+import (
+	"errors"
+
+	osq "github.com/openshift/api/quota/v1"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var ErrNotOpenShift = errors.New("not an OpenShift cluster")
+
+// IsOpenShift detects available endpoints to determine of OpenShift APIs are available
+func (c *APIClient) IsOpenShift() OpenShiftApiLevel {
```

Comment:
Maybe instead this could be `GetOpenShiftGroupVersions() []string` and `IsOpenShift` could just return `true` if any OpenShift group version is available?